samples/modules/thrift/hello_client/prj.conf

```diff
@@ -0,0 +1,75 @@
+# CONFIG_LIB_CPLUSPLUS Dependencies
+CONFIG_NEWLIB_LIBC=y
+CONFIG_NEWLIB_LIBC_NANO=n
+
+# CONFIG_THRIFT Dependencies
+CONFIG_CPP=y
+CONFIG_STD_CPP17=y
+CONFIG_CPP_EXCEPTIONS=y
+CONFIG_EXTERNAL_LIBCPP=y
+CONFIG_POSIX_API=y
+CONFIG_NETWORKING=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETPAIR=y
+CONFIG_HEAP_MEM_POOL_SIZE=16384
+CONFIG_EVENTFD=y
+
+CONFIG_THRIFT=y
+
+CONFIG_TEST_RANDOM_GENERATOR=y
+# pthread_cond_wait() triggers sentinel for some reason
+CONFIG_STACK_SENTINEL=n
+
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_TCP=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_SOCKETS=y
+CONFIG_POSIX_MAX_FDS=6
+CONFIG_NET_CONNECTION_MANAGER=y
+
+# Kernel options
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_INIT_STACKS=y
+
+# Logging
+CONFIG_NET_LOG=y
+CONFIG_LOG=y
+CONFIG_NET_STATISTICS=y
+CONFIG_PRINTK=y
+
+# Network buffers
+CONFIG_NET_PKT_RX_COUNT=16
+CONFIG_NET_PKT_TX_COUNT=16
+CONFIG_NET_BUF_RX_COUNT=64
+CONFIG_NET_BUF_TX_COUNT=64
+CONFIG_NET_CONTEXT_NET_PKT_POOL=y
+
+# IP address options
+CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=3
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=4
+CONFIG_NET_MAX_CONTEXTS=10
+
+# Network application options and configuration
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_CONFIG_NEED_IPV6=y
+CONFIG_NET_CONFIG_MY_IPV6_ADDR="2001:db8::1"
+CONFIG_NET_CONFIG_PEER_IPV6_ADDR="2001:db8::2"
+CONFIG_NET_CONFIG_NEED_IPV4=y
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.0.2.2"
+
+# Number of socket descriptors might need adjusting
+# if there are more than 1 handlers defined.
+CONFIG_POSIX_MAX_FDS=16
+
+# default stack sizes are not enough
```

Comment:
I should clarify, this varies significantly per platform. E.g. `qemu_x86_64`, `qemu_riscv64_smp` require much more than 32-bit platforms due to 64-bit registers but also guard regions for MMU